### PR TITLE
feat: add en_JP locale

### PR DESCRIPTION
**Pull request by revmischa**
Added the amazing Japanese interpretations of American-sounding names from https://docs.google.com/spreadsheets/d/1KOP4mNAX5R0N_dODE6j2gHTK2oTXCIrhAQ0dEniaYFw/edit#gid=0

image

### CODE REVIEW

File: src/locales/en_JP/index.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/*
+ * This file is automatically generated.
+ * Run 'pnpm run generate:locales' to update.
+ */
+import type { LocaleDefinition } from '../..';
+import name_ from './name';
+
+const en_JP: LocaleDefinition = {
+  title: 'English Names - 1994 Fighting Baseball (SNES) Japan Full Rosters',
```
Comment (ST-DDT) [inline src/locales/en_JP/index.ts:9]: I'm not sure about this name.
Maybe `English (Japanese)`?

File: src/locales/en_JP/name/first_name.ts [hunk 2]
```diff
@@ -0,0 +1,703 @@
+export default [
+  'Kevin',
+  'Luis',
+  'Derrick',
```
Comment (ST-DDT) [inline src/locales/en_JP/name/first_name.ts:4]: Are these "westernized" versions of Japanese names?

File: src/locales/en_JP/index.ts (hunk 1 above)
Comment (revmischa) [reply src/locales/en_JP/index.ts:9]: As you wish, but it's just from this one source..

File: src/locale/en_JP.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+const faker = new Faker({
+  locale: 'en_JP',
+  localeFallback: 'en',
+  locales: {
+    en_JP,
+    en,
+  },
+});
+
+export = faker;
```
[the remark this answers is not in the document]
Comment (revmischa) [reply src/locale/en_JP.ts:19]: No I had an old fork, let me export it properly

File: src/locales/en_JP/name/first_name.ts (hunk 2 above)
Comment (revmischa) [reply src/locales/en_JP/name/first_name.ts:4]: https://www.reddit.com/r/baseball/comments/8klxnf/the_full_fighting_baseball_snes_rosters_more_than/

File: src/locales/en_JP/index.ts (hunk 1 above)
Comment (Shinigami92) [reply src/locales/en_JP/index.ts]: Yeah, but maybe... just maybe, someone in the future would like to add something else to the en_JP set and they definitely will not think about this `title` field :wink: 
All out other locales have this not-written-down convention :thinking: 
Something in me tend more to `English (Japanese)`

File: src/locales/en_JP/name/index.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+ * Run 'pnpm run generate:locales' to update.
+ */
+import type { NameDefinitions } from '../../..';
+import first_name from './first_name';
+import last_name from './last_name';
+
+const name = {
+  first_name,
+  last_name,
+} as Partial<NameDefinitions>;
```
Comment (ST-DDT) [inline src/locales/en_JP/name/index.ts:12]: Please run `pnpm run generate:locales` again.

File: src/locales/en_JP/index.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+/*
+ * This file is automatically generated.
+ * Run 'pnpm run generate:locales' to update.
+ */
+import type { LocaleDefinition } from '../..';
+import name_ from './name';
+
+const en_JP: LocaleDefinition = {
+  title: 'English Names (Japan)',
```
Comment (ST-DDT) [inline src/locales/en_JP/index.ts:9]:
```suggestion
  title: 'English (Japan)',
```

@Shinigami92 What do you think?

**Review by ST-DDT — CHANGES_REQUESTED** [review]

File: src/locales/en_JP/name/index.ts (hunk 4 above)
Comment (revmischa) [reply src/locales/en_JP/name/index.ts]: ok done

**Comment by ST-DDT** [comment]
Blocked by #1027 and I cannot push the fix directly to this branch.

**Comment by revmischa** [comment]
> Blocked by #1027 as I cannot push the fix directly to this branch.

I rebased and regenerated

File: src/locales/en_JP/index.ts (hunk 5 above)
Comment (Shinigami92) [reply src/locales/en_JP/index.ts:9]:
```suggestion
  title: 'English (Japan)',
```

**Comment by Shinigami92** [comment]
@revmischa Please use `English (Japan)` instead of `English Names (Japan)` because we use the same of e.g. `English (Ghana)` or `English (India)`
I cannot apply suggestions on your PR, you may have set your branch to not allow maintainers edit

**Review by ST-DDT — COMMENTED** [review]
Let me ask a few more questions regarding how the westernized names work in Japanese.

Is the list you provided (a subset of) a defined list of allowed names that Japanese people can use or is this just a fun/famous list of names in a Japanese context?

**Comment by revmischa** [comment]
> Let me ask a few more questions regarding how the westernized names work in Japanese.
> 
> Is the list you provided (a subset of) a defined list of allowed names that Japanese people can use or is this just a fun/famous list of names in a Japanese context?

The latter. I know of no "allowed list" that exists

**Comment by Shinigami92** [comment]
> The latter. I know of no "allowed list" that exists

In that case, please explain deeply why this is worthfull to be included in Faker. Do you need it for your own app?
Is this needed by others not just your app? Otherwise you also could just use `faker.helpers.arrayElement(yourNameArray)` :thinking:

**Comment by revmischa** [comment]
> > The latter. I know of no "allowed list" that exists
> 
> In that case, please explain deeply why this is worthfull to be included in Faker. Do you need it for your own app? Is this needed by others not just your app? Otherwise you also could just use `faker.helpers.arrayElement(yourNameArray)` 🤔

Because it produces amusing names. Others will enjoy and appreciate it. Same reason we have en_BORK.  https://github.com/faker-js/faker/blob/main/src/locales/en_BORK/lorem/words.ts

**Comment by xDivisionByZerox** [comment]
The problem with intruducing locales is: They are hard to maintain (like really hard). If you introduce this new locale we have to support this best case all features we have. This does not only include names but any module (like interent, finance, etc). If you would currently load an instance of your locale most faker methods would fail (or fall back to english). 

Thats why we have to be very thoughtfull about introducing new locales. 

For existing locales its a bit harder (like en_BORK). They are already out in the wild, we need to keep supporting them unless we drop them. But that would require us to make major releases with not much impact.

Hope that clearyfication helps.

File: src/locales/en_JP/name/first_name.ts [hunk 6]
```diff
@@ -0,0 +1,702 @@
...
+  'Jeff',
+  'Delino',
+  'Tim',
+  'Jarvis',
+  'Robby',
+  'Vince',
+  'Mickey',
+  'Jeff',
+  'Ave',
+  'Bobby',
```
Comment (kid-icarus) [inline src/locales/en_JP/name/first_name.ts:182]: There are many duplicate entries here which is very wasteful and bandwidth intensive. Just search for 'Bob' or 'Bobby'.

**Comment by ST-DDT** [comment]
Some statistics for future reference:
From the firstnames 190 out of 310 are also present in `en`, the remaining 120 names mostly only differ in the spelling of the name.
The lastname look like mutated original names.

These names might be funny when matching first and last names are returned because you can guess the original player name. However randomly shuffled together, they are just random names.
If you think the missing first names are important, then we can consider adding them directly to `en`
For the last names, I'm not sure as I'm not sure whether these are/feel like English last names (I'm probably not the best one to judge that).

File: src/locales/en_JP/name/first_name.ts (hunk 6 above)
Comment (kid-icarus) [reply src/locales/en_JP/name/first_name.ts:182]: Specifically, there are 118 duplicate first names, and 30 duplicate laste names.

**Comment by kid-icarus** [comment]
> These names might be funny when matching first and last names are returned because you can guess the original player name. However randomly shuffled together, they are just random names. If you think the missing first names are important, then we can consider adding them directly to `en` For the last names, I'm not sure whether these are/feel like English last names (I'm probably not the best one to judge that).

As an english speaker I can confirm these are English-sounding last names.

**Comment by revmischa** [comment]
I think random names are fine. I could remove the duplicates.

**Comment by ST-DDT** [comment]
I'm closing this as the additional maintenance efforts outweighs the benefits/fun.

**Pull request closed without merging**